pytest_mh/_private/plugin.py

```diff
@@ -562,6 +566,25 @@ def _clone_function(self, name: str, f: pytest.Function) -> pytest.Function:
             originalname=f.originalname,
         )
 
+    def _preferred_topology(self, item: pytest.Item):
+        if self.mh_ignore_preferred_topology:
+            return
+
+        if len([i for i in item.iter_markers("preferred_topology")]) > 1:
+            raise ValueError("Too many arguments for @pytest.mark.preferred_topology.")
```

Comment:
Hmm, how about taking the mark that is the closest?

Topologies can be inherited from class, this should be possible to inherit as well. I.e.

```python
@pytest.mark.preferred_topology(KnownTopology.LDAP)
@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
class TestFoo():
   def test_bar():
        pass

    @pytest.mark.preferred_topology(KnownTopology.IPA)
    def test_foo():
        pass
```

The preferred topology for test_bar will be LDAP, for test_foo IPA.